### PR TITLE
STT-133_user/couse/enrollment/college/department_endpoints

### PR DESCRIPTION
**Pull request by kaasila**
Still need to test some things, a few endpoints require other people's work (i.e. projects) thus I'm waiting on those.

### CODE REVIEW

File: base/views.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from django.contrib.auth.models import User
```
Comment (ryanlough) [inline base/views.py]: wut is this file for

File: base/department_views.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+from rest_framework import status
+
+from base.department_serializer import DepartmentSerializer
+
+class DepartmentDetail(APIView):
+    def get_object(self, pk):
+        try:
+            return Department.objects.get(pk=pk)
+        except Department.DoesNotExist:
+            raise Http404
```
Comment (ryanlough) [inline base/department_views.py]: You didn't import http404

File: base/course_views.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+from rest_framework import status
+
+from base.course_serializer import CourseSerializer
+
+class CourseDetail(APIView):
+    def get_object(self, pk):
+        try:
+            return Course.objects.get(pk=pk)
+        except Course.DoesNotExist:
+            raise Http404
```
Comment (ryanlough) [inline base/course_views.py]: import http404 or it will blow up D:

File: base/college_views.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+from rest_framework import status
+
+from base.college_serializer import CollegeSerializer
+
+class CollegeDetail(APIView):
+    def get_object(self, pk):
+        try:
+            return College.objects.get(pk=pk)
+        except College.DoesNotExist:
+            raise Http404
```
Comment (ryanlough) [inline base/college_views.py]: http404

File: base/serializers.py [hunk 5]
```diff
@@ -1,32 +1,11 @@
 from django.contrib.auth.models import User
+from base.models import *
 from rest_framework import serializers
 
 """
     THESE ARE JUST EXAMPLES, PLEASE REPLACE
 """
 class UserSerializer(serializers.ModelSerializer):
```
Comment (ryanlough) [inline base/serializers.py]: oh oops, dont use this
use the user serializer in user_serializers.py

Comment (ryanlough) [inline base/serializers.py]: let's delete this file while we're at it

**Comment by ttsapakos** [comment]
now watch me 🐴  🐴

File: base/user_views.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+class UserList(APIView):
+    """
+    List all users, or create a new user.
+    """
+    def get(self, request, format=None):
+        users = User.objects.all()
+        serializer = UserSerializer(users, many=True)
+        return Response(serializer.data)
+
+    def post(self, request, format=None):
```
Comment (ryanlough) [inline base/user_views.py]: wut

**Pull request closed without merging**